Source/src/components/Component.cpp

```diff
@@ -12,6 +12,11 @@ Hachiko::Component::Component(const Type type, GameObject* container, UID id)
     }
 }
 
+bool Hachiko::Component::IsActive() const
+{
+    return active && game_object->IsActive();
```

Comment:
this should not be necessary here, we should validate that the game object is active in the game object itself, thus preventing any lifecycle methods from being called in the component

Reply:
Ur right